fusillade/api/resource/resource_type_name/__init__.py

```diff
@@ -0,0 +1,36 @@
+from flask import request, make_response, jsonify
+
+from fusillade.directory.resource import ResourceType
+from fusillade.utils.authorize import authorize
+
+
+@authorize(["fus:GetResources"], ["arn:hca:fus:*:*:resource/*"])
+def get(token_info: dict, resource_type_name):
+    rt = ResourceType(resource_type_name)
+    rt._exists([rt.name])
+    info = rt.get_info()
+    return make_response(info, 200)
+
+
+@authorize(["fus:PostResources"], ["arn:hca:fus:*:*:resource/*"])
+def post(token_info: dict, resource_type_name):
+    json_body = request.json
+    rt = ResourceType.create(resource_type_name, json_body['actions'])
+    return make_response(f"New resource type {rt.name} created.", 201)
+
+
+@authorize(["fus:PostResources"], ["arn:hca:fus:*:*:resource/*"])
+def delete(token_info: dict, resource_type_name):
+    rt = ResourceType(resource_type_name)
+    rids = rt.list_ids()[0]
+    if rids:
```

Comment:
This test appears to be written expecting `list_ids()` to return a non-empty list containing None when there are no more resources of that type left. Otherwise, getting the `[0]` element of an empty list would raise an exception. Is that the expected behavior?

Would it be better to change this to `if len(rt.list_ids()) > 0`?

Reply:
fixed

Reply:
good catch. TY